UserInterface/Dialogues/IfcTreeItem.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+	Copyright (c) 2021 Technical University of Munich
+	Chair of Computational Modeling and Simulation.
+
+	TUM Open Infra Platform is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License Version 3
+	as published by the Free Software Foundation.
+
+	TUM Open Infra Platform is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "IfcTreeItem.h"
+#include <QStringList>
+#include <type_traits>
+#include <cstdlib>
+#include "Exception\UnhandledException.h"
+#include "visit_struct\visit_struct.hpp"
+
+
+OpenInfraPlatform::UserInterface::IfcTreeItem::IfcTreeItem(std::shared_ptr<OpenInfraPlatform::EarlyBinding::EXPRESSEntity> data, std::shared_ptr<IfcTreeItem> parentItem)
+	:data_(data), parentItem_(parentItem)
+{}
+
+OpenInfraPlatform::UserInterface::IfcTreeItem::~IfcTreeItem()
+{
+	childItems_.clear();
+}
+
+void OpenInfraPlatform::UserInterface::IfcTreeItem::appendchild(std::shared_ptr<IfcTreeItem> child)
+{
+	childItems_.append(child);
+}
+
+std::shared_ptr<OpenInfraPlatform::UserInterface::IfcTreeItem> OpenInfraPlatform::UserInterface::IfcTreeItem::child(int row)
+{
+	if (row < 0 || row >= childItems_.size())
+		return nullptr;
+	return childItems_.at(row);
+}
+
+int OpenInfraPlatform::UserInterface::IfcTreeItem::childCount() const
+{
+	return childItems_.count();
+}
+
+int OpenInfraPlatform::UserInterface::IfcTreeItem::columnCount() const
+{
+	return 3;
+}
+
+QVariant OpenInfraPlatform::UserInterface::IfcTreeItem::data(int row, int column) const
+{
+	//auto attributes = getAttributeDescription();
+
+	//switch (column)
+	//{
+	//case 0:
+	//	return attributes.names_[row];
+	//	break;
+	//case 1:
+	//	return attributes.value_[row];
+	//	break;
+	//case 2:
+	//	return attributes.typename_[row];
+	//	break;
+	//}
+	return QVariant();
+}
+
+int OpenInfraPlatform::UserInterface::IfcTreeItem::row()
+{
+	if (parentItem_)
+	{	
+		return parentItem_->childItems_.indexOf(shared_from_this());
+	}
+	return 0;
+}
+
+std::shared_ptr<OpenInfraPlatform::UserInterface::IfcTreeItem> OpenInfraPlatform::UserInterface::IfcTreeItem::parentItem()
+{
+	if (parentItem_)
+		return parentItem_;
+
+	return std::shared_ptr<IfcTreeItem>();
```

Comment:
This seems weird - why not return a `nullptr`?

Reply:
Resolved: bc8c767